kernel/src/process.rs

```diff
@@ -407,7 +364,7 @@ impl TbfHeader {
         match *self {
             TbfHeader::TbfHeaderV1(_) => mem::size_of::<TbfHeaderV1>() as u32,
             TbfHeader::TbfHeaderV2(hd) =>
-                hd.main.map_or(0, |m| m.protected_size) + hd.base.header_size as u32,
+                hd.main.map_or(0, |m| m.protected_size) + align4!(hd.base.header_size as u32),
```

Comment:
I think you changed this from `align8` to `align4`, but the question from before is the same: shouldn't this already be aligned 4 if this is a valid TBF header?

Reply:
No, but this is a sort of [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle)

Reply:
Hrmm.., yeah.., I see that, though given that properly formed TBF header is somewhat important to application integrity*, perhaps the better model is to fail when the TBF header doesn't match expectations? Could be swayed either way, but I suspect in the case where that field is mis-aligned, there are likely other lurking issues as well.

*I don't think that a mal-formed TBF header could corrupt _other_ applications, I would have to think a bit more if incorrect values could allow an application to do something unexpected.

Reply:
OK, I agree.

> I don't think that a mal-formed TBF header could corrupt other applications

I believe it can only do that if the `total_size` field is incorrect, in which case we might miss subsequent process. But that's more a property of how we're currently account for where processes resides in flash rather than parsing the TBF header.